python-spec/src/somacore/query/query.py

```diff
@@ -241,12 +241,28 @@ def varp(self, layer: str) -> data.SparseRead:
         """
         return self._axisp_inner(_Axis.VAR, layer)
 
+    def obsm(self, layer: str) -> data.SparseRead:
```

Comment:
Argument names are inconsistent between `obsm`, etc and `to_anndata` - one uses `layer`, the other uses `keys`

Noting in case this is not intentional.  IMHO, they would ideally use the same names.   Collection - uses `key`

Reply:
I would settle for `keys`, since that's how AnnData calls them, plus `layers` don't seem to make a lot of sense for embeddings or pairwise matrices (for one, embeddings can have a different number of columns therefore they don't even overlap). The issue is that `obsp` and `varp` were already part of the spec, so renaming the variable to `keys` would be a breaking change. Worth the "risk"?

Reply:
If you want to propose a cleanup, e can fairly easily support both, deprecate one, and remove it in a few releases.

Reply:
If I’m following this discussion correctly, is the concern about renaming the parameters to obsp/varp to `key`? This would *technically* be a break, but only affect anybody who was writing `whatever.varp(layer=xxx)`. I feel pretty comfortable saying this would be fine even in the absence of a transition period.

If we were really concerned, we could also make it a positional-only argument (i.e., use something like `__layer` for the name in old Python, or `def varm(self, /, key: str)` once we jettison v3.7, and end naming worries once and for all.

If I’m misreading, feel free to ignore whatever I just said.

Reply:
Yes, the discussion was about renaming `layer` to `key`. Given that `layer` was, IMHO, a pretty confusing name in the first place for obsp/varp, I'm proposing we just introduce the breaking change and rename it to `key`. @johnkerl any concerns from you?